docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Compiler Command Arguments
+
+The compiler *MUST* receive a new switch, `/editorconfig`, which accepts a single
+file which is an .editorconfig file that may apply to one or more files in the
+compilation. Multiple files are passed with multiple `/editorconfig` switches,
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
```

Comment:
@jaredpar _Aside_ from the issue of path mapping (or more specifically, which paths we choose to map by default) I do not believe we need to prevent the search of parent directories for .editorconfig files by default. While it's possible for a .editorconfig file to affect a build, the conditions under which this would occur are unlikely, and all but impossible to result in someone shipping broken code.

Start with the first two mass filters:

**1. Most users will not have .editorconfig files**

These users are not affected by probing to the drive root.

**2. Of the users who have .editorconfig files, most will not have properties specified in them which affect C# compiler outcome.** Specifically, no properties previously defined on editorconfig.org affect which warnings are reported or the severity with which they are reported during a build

Probing to the drive root may change the set of .editorconfig files passed to the compiler on a local machine, but it will not change the build outcome (success/fail) under any circumstances.

Now we get a bit more specific.

**3. Most users will not have analyzers installed via NuGet**

For this category, even a custom pathological **.editorconfig** is only capable of changing the severity of warnings reported by the C# compiler itself. To break a build on a local machine, one would have to explicitly increase the severity of a C# compiler warning via a .editorconfig file which is not part of the project.

**4. Most users who customize .editorconfig in ways that affect builds will do so in the context of particular projects**

When **.editorconfig** is added to a specific project with root=true, searching to the drive root by default is not a problem.

**5. For the remaining user who wants to customize a build-affecting non-project-specific .editorconfig for some reason, they probably have a reason to do so and we have no real need to prevent them from doing so.**